roles/wordpress-install/tasks/main.yml

```diff
@@ -25,43 +25,30 @@
   changed_when: "'Nothing to install or update' not in composer_results.stderr"
 
 - name: Install WP
-  command: wp core install
-           --allow-root
-           --url="{{ site_env.wp_home }}"
-           --title="{{ item.value.site_title | default(item.key) }}"
-           --admin_user="{{ item.value.admin_user | default('admin') }}"
-           --admin_password="{{ vault_wordpress_sites[item.key].admin_password }}"
-           --admin_email="{{ item.value.admin_email }}"
-  args:
-    chdir: "{{ www_root }}/{{ item.key }}/{{ item.value.current_path | default('current') }}/"
-  register: wp_install_results
-  with_dict: "{{ wordpress_sites }}"
-  when: item.value.site_install | default(true) and not item.value.multisite.enabled | default(false)
-  changed_when: "'WordPress is already installed.' not in wp_install_results.stdout"
-
-- name: Install WP Multisite
-  command: wp core multisite-install
+  command: wp core {{ item.value.multisite.enabled | default(false) | ternary('multisite-install', 'install') }}
            --allow-root
            --url="{{ site_env.wp_home }}"
+           {% if item.value.multisite.enabled | default(false) %}
            --base="{{ item.value.multisite.base_path | default('/') }}"
            --subdomains="{{ item.value.multisite.subdomains | default('false') }}"
+           {% endif %}
            --title="{{ item.value.site_title | default(item.key) }}"
            --admin_user="{{ item.value.admin_user | default('admin') }}"
            --admin_password="{{ vault_wordpress_sites[item.key].admin_password }}"
            --admin_email="{{ item.value.admin_email }}"
   args:
     chdir: "{{ www_root }}/{{ item.key }}/{{ item.value.current_path | default('current') }}/"
-  register: wp_multisite_install_results
+  register: wp_install
   with_dict: "{{ wordpress_sites }}"
-  when: item.value.site_install | default(true) and item.value.multisite.enabled | default(false)
-  changed_when: "'The network already exists.' not in wp_multisite_install_results.stdout"
+  when: item.value.site_install | default(true)
+  changed_when: "'WordPress is already installed.' not in wp_install.stdout and 'The network already exists.' not in wp_install.stdout"
 
 - name: Setup Permalink Structure
-  command: wp rewrite structure {{ item.value.initial_permalink_structure | default("/%postname%/") }} --allow-root
+  command: wp rewrite structure {{ wordpress_sites[item.item.key].initial_permalink_structure | default("/%postname%/") }} --allow-root
```

Comment:
I'm going to push a commit that changes the two instances of `wordpress_sites[item.item.key]` to `item.item.value`. That should be simpler and more closely resemble the existing code. We'll want to squash before merging.

I failed to consider that the values needed were contained in the registered variable, so we don't need to call on the `wordpress_sites` dict.